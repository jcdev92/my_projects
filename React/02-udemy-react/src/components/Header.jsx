import React from 'react';
import PropTypes from 'prop-types';

// PropTypes are used to validate the data type of the props passed to the component

function Header({title, subtitle}) {
    return (
        <section>
            <h1 data-testid="test-title">{title}</h1>
            <h2>{subtitle}</h2>
            <h2>{subtitle}</h2>
            <h2>{subtitle}</h2>
        </section>
    );
}

Header.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.number.isRequired
}

// Default props are used to set default values for the props if they are not passed to the component

Header.defaultProps = {
    title: 'Default title',
    subtitle: 0
}

export default Header;