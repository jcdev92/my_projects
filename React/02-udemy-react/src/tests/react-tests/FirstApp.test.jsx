import {render} from "@testing-library/react";
import Header from "../../components/Header.jsx";


describe('Pruebas en <Header/>', () => {
        test('debe de hacer match con el snapshot', () => {
            const {container} = render(<Header title="Hello World" subtitle={15}/>);
            expect(container).toMatchSnapshot();
        })

        test('debe de mostrar el titulo enviado por props', () => {
            const title = 'Hello World';
            const subtitle = 15;
            const {getByText, getByTestId} = render(<Header title="Hello World" subtitle={15}/>);

            // expect(getByText('Hello World')).toBeTruthy();;

            const byTestId = getByTestId('test-title');
            expect(byTestId.innerHTML).toBe(title);
            expect(getByText(subtitle)).toBeTruthy();


        })
})