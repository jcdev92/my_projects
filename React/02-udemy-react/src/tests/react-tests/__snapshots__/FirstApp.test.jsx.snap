// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Header/> debe de hacer match con el snapshot 1`] = `
<div>
  <section>
    <h1
      data-testid="test-title"
    >
      Hello World
    </h1>
    <h2>
      15
    </h2>
  </section>
</div>
`;
